test/conformance/helpers/broker_data_plane_test_helper.go

```diff
@@ -0,0 +1,458 @@
+		client := lib.Setup(t, true, setupClient...)
+		defer lib.TearDown(client)
+
+		brokerName := "br"
+		triggerName := "trigger"
+		secondTriggerName := "second-trigger"
+		loggerName := "logger-pod"
+		secondLoggerName := "second-logger-pod"
+		eventTracker, _ := recordevents.StartEventRecordOrFail(client, loggerName)
+		defer eventTracker.Cleanup()
+		secondTracker, _ := recordevents.StartEventRecordOrFail(client, secondLoggerName)
+		defer secondTracker.Cleanup()
+		client.WaitForAllTestResourcesReadyOrFail()
+
+		config := client.CreateBrokerConfigMapOrFail(brokerName, &channel)
+
+		broker := client.CreateBrokerV1Beta1OrFail(
+			brokerName,
+			resources.WithBrokerClassForBrokerV1Beta1(brokerClass),
+			resources.WithConfigForBrokerV1Beta1(config),
+		)
+		client.WaitForResourceReadyOrFail(broker.Name, testlib.BrokerTypeMeta)
+
+		trigger := client.CreateTriggerOrFailV1Beta1(
+			triggerName,
+			resources.WithBrokerV1Beta1(broker.Name),
+			resources.WithAttributesTriggerFilterV1Beta1(eventingv1beta1.TriggerAnyFilter, eventingv1beta1.TriggerAnyFilter, nil),
+			resources.WithSubscriberServiceRefForTriggerV1Beta1(loggerName),
+		)
+
+		client.WaitForResourceReadyOrFail(trigger.Name, testlib.TriggerTypeMeta)
+		secondTrigger := client.CreateTriggerOrFailV1Beta1(
+			secondTriggerName,
+			resources.WithBrokerV1Beta1(broker.Name),
+			resources.WithAttributesTriggerFilterV1Beta1("filtered-event", eventingv1beta1.TriggerAnyFilter, nil),
+			resources.WithSubscriberServiceRefForTriggerV1Beta1(secondLoggerName),
+		)
+		client.WaitForResourceReadyOrFail(secondTrigger.Name, testlib.TriggerTypeMeta)
+		eventID := "consumer-broker-tests"
+		baseSource := "consumer-test-sender"
+		baseEvent := ce.NewEvent()
+		baseEvent.SetID(eventID)
+		baseEvent.SetType(testlib.DefaultEventType)
+		baseEvent.SetSource(baseSource)
+		baseEvent.SetSpecVersion("1.0")
+		body := fmt.Sprintf(`{"msg":%q}`, eventID)
+		if err := baseEvent.SetData(ce.EncodingStructured.String(), []byte(body)); err != nil {
+			t.Fatalf("Cannot set the payload of the baseEvent: %s", err.Error())
+		}
+
+		t.Run("No upgrade of version", func(t *testing.T) {
+			event := baseEvent
+			source := "no-upgrade"
+			event.SetID(source)
+			event.Context = event.Context.AsV03()
+
+			client.SendEventToAddressable(source+"-sender", broker.Name, testlib.BrokerTypeMeta, event)
+			originalEventMatcher := recordevents.MatchEvent(cetest.AllOf(
+				cetest.HasSpecVersion("0.3"),
+				cetest.HasId("no-upgrade"),
+			))
+			eventTracker.AssertExact(1, originalEventMatcher)
+
+		})
+
+		t.Run("Attributes received should be the same as produced (attributes may be added)", func(t *testing.T) {
+			event := baseEvent
+			id := "identical-attibutes"
+			event.SetID(id)
+			client.SendEventToAddressable(id+"-sender", broker.Name, testlib.BrokerTypeMeta, event)
+			originalEventMatcher := recordevents.MatchEvent(cetest.AllOf(
```

Comment:
you don't need that all of, `MatchEvent` already accepts a variadic which then pass to `AllOf`